source/common/router/router.cc

```diff
@@ -453,6 +453,10 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::RequestHeaderMap& headers,
 
   // Set up stat prefixes, etc.
   request_vcluster_ = route_entry_->virtualCluster(headers);
+  if (request_vcluster_) {
+    const std::string& vc_name = request_vcluster_->vcName();
+    callbacks_->streamInfo().setVirtualClusterName(vc_name);
```

Comment:
can we depend on the requst_vcluster_ outlive the stream-info? If so let's have this be a reference.

Note that the StatName() (which you previously were converting to a string) is already essentially a reference to storage held elsewhere, like a string_view or `string&`, so if we parallel the ownership of that we should be able to hold a reference to a string with a similar lifetime.

Reply:
@jmarantz I'm not 100% sure (not very familiar) that it'd be safe to store a reference instead of creating a Deep Copy. But if you think it's okay then I can change it.

Reply:
I don't have a good feel for it. If you don't either you could just leave a comment for now to investigate and do a deep copy.

Reply:
I changed it to do a deep copy instead for now and left a comment/TODO to investigate more for whether it could be a reference.